frontend/src/components/SignUp/SignUpForm.jsx

```diff
@@ -0,0 +1,172 @@
+    this.setState({[e.target.name]: e.target.value});
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const {name, email, password} = this.state;
+    const formData = {name, email, password};
+    const requestConfig = {
+      headers: new Headers({"Content-Type": "application/json"}),
+      method: "POST",
+      body: JSON.stringify(formData),
+    };
+    fetch("http://localhost:6543/api/sign-up", requestConfig)
+      .then(response => response.json())
+      .then(response => this.setState({serverResponse: response.success}))
+      .catch(error => this.setState({errors: true}));
+  }
+
+  render() {
+    const {classes} = this.props;
+    if (this.state.errors) {
+      return (
+        <SubmittedForm classes={classes} message={"Ooops something went wrong! Please try again."}/>
+
+      );
+    } else if (this.state.serverResponse) {
+      return (
+        <SubmittedForm classes={classes} message={"Thanks for registration!"}/>
+      );
+    } else {
+      return (
+        <Grid container justify='center' className={classes.form}>
+          <Card className={classes.cardBody}>
+            <Typography variant="h4" align='center'>
+              Create your account
+            </Typography>
+            <ValidatorForm ref="form" onSubmit={this.handleSubmit} debounceTime={2000}>
+              <Grid container direction='column' alignContent='center'>
+                <CardContent>
+                  <Grid item>
+                    <TextValidator
+                      label="Name"
+                      onChange={this.handleChange}
+                      className={classes.textField}
+                      validators={["required", "minStringLength:3", "trim"]}
+                      errorMessages={[
+                        "Name is required",
+                        "Name must have at least 3 characters",
+                        "You didn't enter any character"
+                        ]}
+                      value={this.state.name}
+                      margin="normal"
+                      name="name"
+                    />
+                    <TextValidator
+                      label="Email"
+                      onChange={this.handleChange}
+                      className={classes.textField}
+                      validators={['required', 'isEmail']}
+                      errorMessages={[
+                        "Email is required",
+                        "Email is not valid"
+                        ]}
+                      value={this.state.email}
+                      margin="normal"
+                      name="email"
+                    />
+                  </Grid>
+                  <Grid item>
+                    <TextValidator
+                      label="Password"
+                      onChange={this.handleChange}
+                      className={classes.textField}
+                      validators={['required', "minStringLength:8", "matchRegexp:^.*[a-zA-Zа-яА-Я]{2}"]}
```

Comment:
Please try to avoid using non-ascii symbols in code. There is recommendation from MDN how to use unicode symbols in  regexps: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#Regular_expression_and_Unicode_characters

Reply:
Thanks for the information. I want to make some changes:

1. I want to use this RegExp `.*[a-zA-Z\u0430-\u044f\u0410-\u042f].*` (any characters in the password, but must contain at least one letter).
2. I want to edit restriction for name (2 characters instead of 3)
3. I want to show error after unsuccessful form submission on the same page.
4. I want to replace ordinary functions with arrows functions.

What do you think about it all?

Reply:
Does we need restrict min length on name? yesterday, on demo, we decide just make it field required as i understood

Reply:
@viktormovchaniuk yep, you are right

Reply:
> Does we need restrict min length on name? yesterday, on demo, we decide just make it field required as i understood

Because of our Chinese users, where one symbol can mean one sentence. But I thought we don't orient at them.

Reply:
I already removed min length restriction.

Be cool, be nice
Removed restrictions, guys